### PR TITLE
Add presets for various jobs that use common volumes and env

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
for the task https://github.ibm.com/powercloud/container-dev/issues/1589

### CODE REVIEW

**Comment by Rajalakshmi-Girish** [comment]
Have done a test run with the new preset enabled. The job ran as expected.
https://prow.ppc64le-cloud.org/view/gs/ppc64le-kubernetes/logs/test-periodic-kubernetes-containerd-conformance-test-ppc64le/1591131451661750272
@mkumatag Can you PTAL at this PR?

File: config/jobs/periodic/kubernetes/perf-tests/perf-tests-kubernetes-periodics.yaml [hunk 1]
```diff
@@ -99,22 +91,10 @@ periodics:
...
     - image: quay.io/powercloud/all-in-one:0.5
       command:
       - /bin/bash
-      volumeMounts:
-      - mountPath: /etc/secret-volume
-        name: powercloud-bot-key
-        readOnly: true
-      envFrom:
-      - secretRef:
-          name: ibm-cloud-credentials
```
Comment (mkumatag) [inline config/jobs/periodic/kubernetes/perf-tests/perf-tests-kubernetes-periodics.yaml:117]: this is not covered in the `preset-ssh-bot` preset, can you ptal

File: config/jobs/periodic/kubernetes/test-kubernetes-periodics.yaml [hunk 2]
```diff
@@ -39,8 +39,10 @@ periodics:
               rm -rf /usr/local/go
               tar -C /usr/local -xzf go.tgz
               make test KUBE_RACE=-race
-  - name: periodic-kubernetes-containerd-conformance-test-ppc64le
+  - name: test-periodic-kubernetes-containerd-conformance-test-ppc64le
```
Comment (mkumatag) [inline config/jobs/periodic/kubernetes/test-kubernetes-periodics.yaml:42]: any specific reason for this `test-` prefix part of this PR?

File: config/jobs/ppc64le-cloud/all-in-one/allinone-postsubmit.yaml [hunk 3]
```diff
@@ -13,15 +13,6 @@ postsubmits:
         serviceAccountName: build-img
         containers:
           - image: quay.io/powercloud/all-in-one:0.5
-            env:
```
Comment (mkumatag) [inline config/jobs/ppc64le-cloud/all-in-one/allinone-postsubmit.yaml:16]: I don't see any preset added here for this job after removing the volumes!

File: config/jobs/ppc64le-cloud/build-docker/postsubmit-build-docker.yaml [hunk 4]
```diff
@@ -135,15 +127,3 @@ postsubmits:
...
-            mountPath: /var/lib/docker
-          env:
-            - name: S3_SECRET_AUTH
-              valueFrom:
-                secretKeyRef:
-                  name: docker-s3-credentials
-                  key: password
-        volumes:
-        - name: docker-root
-          emptyDir: {}
```
Comment (mkumatag) [inline config/jobs/ppc64le-cloud/build-docker/postsubmit-build-docker.yaml:149]: preset is missing in this job for this removal

File: config/jobs/ppc64le-cloud/build-docker/postsubmit-test-repo-docker.yaml [hunk 5]
```diff
@@ -115,15 +104,3 @@ postsubmits:
 
           securityContext:
             privileged: true
-          volumeMounts:
```
Comment (mkumatag) [inline config/jobs/ppc64le-cloud/build-docker/postsubmit-test-repo-docker.yaml:118]: preset is missing for this job

File: config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml [hunk 6]
```diff
@@ -15,15 +15,6 @@ postsubmits:
         serviceAccountName: build-img
         containers:
           - image: quay.io/powercloud/all-in-one:0.5
-            env:
```
Comment (mkumatag) [inline config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml:18]: preset is missing here

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: config/jobs/periodic/kubernetes/perf-tests/perf-tests-kubernetes-periodics.yaml (hunk 1 above)
Comment (Rajalakshmi-Girish) [reply config/jobs/periodic/kubernetes/perf-tests/perf-tests-kubernetes-periodics.yaml:117]: The `preset-ssh-bot` definition has the environment variable `TF_VAR_powervs_api_key` fetched from `ibm-cloud-credentials` secret.

Though the secret `ibm-cloud-credentials` has two env variables namely `TF_VAR_powervs_api_key` and `IBMCLOUD_API_KEY`, only the first one is needed by the jobs where we are replacing the presets.

Also, the tag `envFrom` cannot be used in `presets` definition. Hence we are using `env:` and fetching only the required environment variable.

File: config/jobs/periodic/kubernetes/test-kubernetes-periodics.yaml (hunk 2 above)
Comment (Rajalakshmi-Girish) [reply config/jobs/periodic/kubernetes/test-kubernetes-periodics.yaml:42]: Hmm, it was added by mistake. Will remove it in the next patch.

File: config/jobs/ppc64le-cloud/all-in-one/allinone-postsubmit.yaml (hunk 3 above)
Comment (Rajalakshmi-Girish) [reply config/jobs/ppc64le-cloud/all-in-one/allinone-postsubmit.yaml:16]: Missed adding. Will do in next patch.

File: config/jobs/ppc64le-cloud/build-docker/postsubmit-build-docker.yaml (hunk 4 above)
Comment (Rajalakshmi-Girish) [reply config/jobs/ppc64le-cloud/build-docker/postsubmit-build-docker.yaml:149]: Missed adding. Will do it in the next patch.

File: config/jobs/ppc64le-cloud/build-docker/postsubmit-test-repo-docker.yaml (hunk 5 above)
Comment (Rajalakshmi-Girish) [reply config/jobs/ppc64le-cloud/build-docker/postsubmit-test-repo-docker.yaml:118]: Adding it in next patch.

File: config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml (hunk 6 above)
Comment (Rajalakshmi-Girish) [reply config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml:18]: Will add in next patch.

**Review by Rajalakshmi-Girish — COMMENTED** [review]
PTAL @mkumatag

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**